avm/res/app/container-app/main.bicep

```diff
@@ -301,3 +299,125 @@ type roleAssignmentType = {
   @description('Optional. The Resource Id of the delegated managed identity resource.')
   delegatedManagedIdentityResourceId: string?
 }[]?
+
+type container = {
+  @description('Optional. Container start command arguments.')
+  args: string[]?
+
+  @description('Optional. Container start command.')
+  command: string[]?
+
+  @description('Optional. Container environment variables.')
+  env: environmentVar[]?
+
+  @description('Required. Container image tag.')
+  image: string
+
+  @description('Optional. Custom container name.')
+  name: string?
+
+  @description('Optional. List of probes for the container')
+  probes: containerAppProbe[]?
+
+  @description('Required. Container resource requirements.')
+  resources: object
+
+  @description('Optional. Container volume mounts.')
+  volumeMounts: volumeMount[]?
+}
+
+type environmentVar = {
+  @description('Required. Environment variable name.')
+  name: string
+
+  @description('Optional. Name of the Container App secret from which to pull the environment variable value.')
+  secretRef: string?
+
+  @description('Optional. Non-secret environment variable value.')
+  value: string?
+}
+
+type containerAppProbe = {
+  @description('Optional. Minimum consecutive failures for the probe to be considered failed after having succeeded. Defaults to 3.')
+  @minValue(1)
+  @maxValue(10)
+  failureThreshold: int?
+
+  @description('Optional. HTTPGet specifies the http request to perform.')
+  httpGet: containerAppProbeHttpGet?
+
+  @description('Optional. Number of seconds after the container has started before liveness probes are initiated.')
+  @minValue(1)
+  @maxValue(60)
+  initialDelaySeconds: int?
+
+  @description('Optional. How often (in seconds) to perform the probe. Default to 10 seconds.')
+  @minValue(1)
+  @maxValue(240)
+  periodSeconds: int?
+
+  @description('Optional. Minimum consecutive successes for the probe to be considered successful after having failed. Defaults to 1. Must be 1 for liveness and startup.')
+  @minValue(1)
+  @maxValue(10)
+  successThreshold: int?
+
+  @description('Optional. TCPSocket specifies an action involving a TCP port. TCP hooks not yet supported.')
+  tcpSocket: containerAppProbeTcpSocket?
+
+  @description('Optional. Optional duration in seconds the pod needs to terminate gracefully upon probe failure. The grace period is the duration in seconds after the processes running in the pod are sent a termination signal and the time when the processes are forcibly halted with a kill signal. Set this value longer than the expected cleanup time for your process. If this value is nil, the pod\'s terminationGracePeriodSeconds will be used. Otherwise, this value overrides the value provided by the pod spec. Value must be non-negative integer. The value zero indicates stop immediately via the kill signal (no opportunity to shut down). This is an alpha field and requires enabling ProbeTerminationGracePeriod feature gate. Maximum value is 3600 seconds (1 hour)')
```

Comment:
```suggestion
  @description('Optional. Optional duration in seconds the pod needs to terminate gracefully upon probe failure. The grace period is the duration in seconds after the processes running in the pod are sent a termination signal and the time when the processes are forcibly halted with a kill signal. Set this value longer than the expected cleanup time for your process. If this value is nil, the pod\'s terminationGracePeriodSeconds will be used. Otherwise, this value overrides the value provided by the pod spec. Value must be non-negative integer. The value zero indicates stop immediately via the kill signal (no opportunity to shut down). This is an alpha field and requires enabling ProbeTerminationGracePeriod feature gate. Maximum value is 3600 seconds (1 hour).')
```